serviceAccessConfig/lib/serviceAccessConfig/accessrulegenerator.py

```diff
@@ -0,0 +1,186 @@
+
+import ConfigParser
+import hashlib
+import logging
+import os
+import requests
+import sys
+import threading
+
+from generatorexceptions import *
+
+
+class ServiceAccessGenerator(object):
+    """Base class for any plugin that implements access rules generation
+       for a specific service"""
+
+    # ======================================================================
+    def __init__(self, ip_source_config_file_name):
+        self.ip_source_config_file_name = ip_source_config_file_name
+        self.ip_source_config_file_sha1 = None
+
+    # ======================================================================
+    def set_config_values(self, service_access_config, section_name=None):
+        """Load the configuration file for the rules generator"""
+
+        base_error_msg = 'Configuration error. %s must be specified with the '
+        base_error_msg += '"%s" option in the "%s" section of the config file.'
+
+        if not section_name:
+            section_name = self.section_name
+        self.service_name = section_name
+        if service_access_config.has_option(section_name, 'serviceName'):
+            self.service_name = service_access_config.get(
+                section_name,
+                'serviceName'
+            )
+
+        if not service_access_config.has_option(section_name, 'serviceConfig'):
+            error_msg = base_error_msg % (
+                'A configuration file to modify',
+                'serviceConfig',
+                section_name
+            )
+            logging.error(error_msg)
+            raise ServiceAccessGeneratorConfigError(error_msg)
+
+        self.service_config = service_access_config.get(
+            section_name,
+            'serviceConfig'
+        )
+
+        self.request = None
+        if service_access_config.has_option(section_name, 'request'):
+            self.request = service_access_config.get(section_name, 'request')
+
+        if not service_access_config.has_option(
+                section_name,
+                'updateInterval'):
+            error_msg = base_error_msg % (
+                'An update interval',
+                'updateInterval',
+                section_name
+            )
+            logging.error(error_msg)
+            raise ServiceAccessGeneratorConfigError(error_msg)
+
+        self.interval = service_access_config.get(
+            section_name,
+            'updateInterval'
+        )
+
+        return 1
+
+    # ======================================================================
+    def update_config(self):
+        """Update the Apache configuration file"""
+        ctx = open(self.ip_source_config_file_name, 'r').read()
+        sha1 = hashlib.sha1()
+        sha1.update(ctx)
+        if (
+                not self.ip_source_config_file_sha1 or
+                self.ip_source_config_file_sha1 != sha1.digest()
+        ):
+            self.ip_source_config_file_sha1 = sha1.digest()
```

Comment:
Nope, it stays in memory only, i.e. on restart of this service we will always force a restart of the dependent service.